packages/integrations/node/src/preview.ts

```diff
@@ -33,49 +26,19 @@ const preview: CreatePreviewServer = async function ({
 		if ((err as any).code === 'ERR_MODULE_NOT_FOUND') {
 			throw new AstroError(
 				`The server entrypoint ${fileURLToPath(
-					serverEntrypoint
+					preview.serverEntrypoint
 				)} does not exist. Have you ran a build yet?`
 			);
 		} else {
 			throw err;
 		}
 	}
-
-	const handler: http.RequestListener = (req, res) => {
-		ssrHandler(req, res);
-	};
-
-	const baseWithoutTrailingSlash: string = base.endsWith('/')
-		? base.slice(0, base.length - 1)
-		: base;
-	function removeBase(pathname: string): string {
-		if (pathname.startsWith(base)) {
-			return pathname.slice(baseWithoutTrailingSlash.length);
-		}
-		return pathname;
-	}
-
-	const server = createServer(
-		{
-			client,
-			port,
-			host,
-			removeBase,
-			assets: options.assets,
-		},
-		handler
-	);
-	const address = getNetworkAddress('http', host, port);
-
-	if (host === undefined) {
-		logger.info(
-			`Preview server listening on \n  local: ${address.local[0]} \t\n  network: ${address.network[0]}\n`
-		);
-	} else {
-		logger.info(`Preview server listening on ${address.local[0]}`);
-	}
-
+	const host = preview.host ?? "127.0.0.1"
+	const port = preview.port ?? 4321
+	const server = createServer(ssrHandler, host, port);
+	logListeningOn(preview.logger, server.server, options)
+	server.server.listen(port, host);
```

Comment:
This causes a race in the tests on some cases and make environment problems preventing the server from starting fail with timeouts hiding the error.

```suggestion
	await new Promise<void>((resolve, reject) => {
	    server.once('listening', resolve);
	    server.once('error', reject);
	    server.server.listen(port, host);
	});
```

Reply:
That's perfect, thanks!